testssl.sh

```diff
@@ -2909,37 +2910,51 @@ run_cipher_match(){
+
+          for proto in $protos_to_try; do
+               if [[ "$proto" == "-tls1_1" ]]; then
+                    num_bundles=1
+                    bundle_size=$nr_ossl_ciphers
+               fi
+               for (( bundle=0; bundle < num_bundles; bundle++ )); do
+                    end_of_bundle=$bundle*$bundle_size+$bundle_size
+                    [[ $end_of_bundle -gt $nr_ossl_ciphers ]] && end_of_bundle=$nr_ossl_ciphers
+                    while true; do
+                         ciphers_to_test=""
+                         for (( i=bundle*bundle_size; i < end_of_bundle; i++ )); do
+                              ! "${ciphers_found2[i]}" && ciphers_to_test+=":${ciph2[i]}"
+                         done
+                         [[ -z "$ciphers_to_test" ]] && break
+                         $OPENSSL s_client $(s_client_options "$proto -cipher "${ciphers_to_test:1}" $STARTTLS $BUGS -connect $NODEIP:$PORT $PROXY $SNI") >$TMPFILE 2>$ERRFILE </dev/null
+                         sclient_connect_successful "$?" "$TMPFILE" || break
+                         cipher=$(get_cipher $TMPFILE)
+                         [[ -z "$cipher" ]] && break
+                         for (( i=bundle*bundle_size; i < end_of_bundle; i++ )); do
+                              [[ "$cipher" == "${ciph2[i]}" ]] && ciphers_found2[i]=true && break
+                         done
+                         [[ $i -eq $end_of_bundle ]] && break
+                         i=${index[i]}
+                         ciphers_found[i]=true
+                         [[ "$cipher" == TLS13* ]] && kx[i]="$(read_dhtype_from_file $TMPFILE)"
+                         if [[ ${kx[i]} == "Kx=ECDH" ]] || [[ ${kx[i]} == "Kx=DH" ]] || [[ ${kx[i]} == "Kx=EDH" ]]; then
+                              dhlen=$(read_dhbits_from_file "$TMPFILE" quiet)
+                              kx[i]="${kx[i]} $dhlen"
+                         fi
+                         "$SHOW_SIGALGO" && grep -q "\-\-\-\-\-BEGIN CERTIFICATE\-\-\-\-\-" $TMPFILE && \
+                              sigalg[i]="$(read_sigalg_from_file "$TMPFILE")"
                     done
-                    [[ $i -eq $end_of_bundle ]] && break
-                    i=${index[i]}
-                    ciphers_found[i]=true
-                    if [[ ${kx[i]} == "Kx=ECDH" ]] || [[ ${kx[i]} == "Kx=DH" ]] || [[ ${kx[i]} == "Kx=EDH" ]]; then
-                         dhlen=$(read_dhbits_from_file "$TMPFILE" quiet)
-                         kx[i]="${kx[i]} $dhlen"
-                    fi
-                    "$SHOW_SIGALGO" && grep -q "\-\-\-\-\-BEGIN CERTIFICATE\-\-\-\-\-" $TMPFILE && \
-                         sigalg[i]="$(read_sigalg_from_file "$TMPFILE")"
                done
           done
 
           if "$using_sockets"; then
                for (( i=0; i < nr_ciphers; i++ )); do
-                    if ! "${ciphers_found[i]}" && [[ "${sslvers[i]}" != "SSLv2" ]] && [[ "${hexcode[i]}" != "13"* ]]; then
```

Comment:
Quick q: why was the ``${hexcode[i]}" != "13"* `` in before?

Reply:
To exclude any TLSv1.3 ciphers, since previously TLSv1.3 ciphers weren't being tested. I think the reason this was included here and not in other functions (e.g., `run_all_ciphers()`) is that at the time this line of code was added (back when `run_cipher_match()` was called `test_just_one()`) the output was always showing each tested cipher, not just those that were available, even if the `--show-each` flag was not set.